poly-synth/src/poly-synth.js

```diff
@@ -103,15 +115,47 @@ class PolySynth {
     let reverbInputNode = new GainNode(this.context_);
     let reverbOutputNode = this.createReverbGraph_(reverbInputNode);
 
-    // The client is responsible for connecting |this.output| to a destination.
-    this.output = new GainNode(this.context_);
-
-    this.voiceOutput_.connect(bitcrusherInputNode);
+    this.voiceOutput_.connect(noisegateInputNode);
+    noisegateOutputNode.connect(bitcrusherInputNode);
     bitcrusherOutputNode.connect(delayInputNode);
     delayOutputNode.connect(reverbInputNode);
     reverbOutputNode.connect(this.output);
   }
 
+  createNoisegateSideChainGraph_(inputNode) {
+    this.noisegate = new NoiseGateSideChain(this.context_, {
+      numberOfChannels: 2,
+      attack: this.noisegateAttack,
+      release: this.noisegateRelease,
+      threshold: this.threshold
+    });
+
+    this.drumSource_ = new AudioBufferSourceNode(this.context_);
+    this.activeNoisegateRoute = new GainNode(this.context_, {gain: 0});
+    this.bypassNoisegateRoute = new GainNode(this.context_, {gain: 1});
+    this.synthAndDrumMerger_ =
+        new ChannelMergerNode(this.context_, {numberOfInputs: 2});
+    let noisegateOutputNode = new GainNode(this.context_);
+
+    inputNode.connect(this.bypassNoisegateRoute).connect(noisegateOutputNode);
+    
+    // The noise gate will build an envelope on the output of the drum sample
+    // (in channel 0) but the noise gate will modify the output of the
+    // synthesizer (channel 1).
+    this.drumSource_.connect(this.synthAndDrumMerger_, 0, 0);
+    inputNode.connect(this.synthAndDrumMerger_, 0, 1);
+    this.synthAndDrumMerger_.connect(this.noisegate.input);
+    this.noisegate.output.connect(
+        this.activeNoisegateRoute.connect(noisegateOutputNode));
+
+    // If |this.drumVolume| is positive, the drum sample is audible but bypasses
+    // the other synthesizer effects.
```

Comment:
Then it's not really part of the side chain, then, I think.  It's really a gain knob for the beat track. Not sure how to address this in the app.